integration-tests/baas-test-server/index.ts

```diff
@@ -90,223 +69,144 @@ function ensureDocker() {
   }
 }
 
-function getHeadCommit(cwd: string) {
-  return execSync("git rev-parse HEAD", { cwd, encoding: "utf8" }).trim();
-}
-
-function getGoVersion() {
-  return execSync("go version", { encoding: "utf8" }).trim();
-}
-
-function ensureBaasRepo() {
-  try {
-    const commit = getHeadCommit(baasPath);
-    console.log(`Using BaaS ${commit}`);
-  } catch (err) {
-    if (err instanceof Error && err.message.includes("ENOENT")) {
-      // Clone the baas repository
-      execSync(`git clone --depth=1 ${BAAS_REPO} baas`, {
-        stdio: "inherit",
-      });
-    } else {
-      throw err;
-    }
+function ensureNoBaas() {
+  const container = execSync(`docker ps -q --filter "name=${BAAS_CONTAINER_NAME}"`, { encoding: "utf8" }).trim();
+  if (container) {
+    console.log(`Killing existing container (id = ${container})`);
+    execSync(`docker kill ${container}`);
   }
 }
 
-function ensureBaasUIRepo() {
-  if (!fs.existsSync(staticPath)) {
-    console.log("Missing the static directory - creating!");
-    fs.mkdirSync(staticPath, { recursive: true });
-  }
-  try {
-    const commit = getHeadCommit(staticAppPath);
-    console.log(`Using BaaS UI ${commit}`);
-  } catch (err) {
-    if (err instanceof Error && err.message.includes("ENOENT")) {
-      // Clone the baas ui repository
-      execSync(`git clone --depth=1 ${BAAS_UI_REPO} app`, {
-        cwd: staticPath,
-        stdio: "inherit",
-      });
-    } else {
-      throw err;
+type Image = {
+  buildVariant: string;
+  imageTag: string;
+  /*
+  _id: string;
+  project: string;
+  order: number;
+  versionId: string;
+  taskId: string;
+  execution: number;
+  timestamp: string;
+  branch: string;
+  revision: string;
+  */
+};
+
+function assertImage(value: unknown): asserts value is Image {
+  assert(typeof value === "object" && value !== null);
```

Comment:
```suggestion
  assert(typeof value === "object" && value !== null && !Array.isArray(value));
```

Reply:
I think it's fine. Without the explicit check for array, the property checks below will return undefined and fail their assertions.